File: frontend/src/microservices/advertising/objects/GenerateAdvertisementImageCandidatesResponse.ts
// 本文件定义 advertising 模块的 `GenerateAdvertisementImageCandidatesResponse`，作为图片候选响应并提供 JSON 编解码。

import type { AdvertisementImageCandidateResponse } from './AdvertisementImageCandidateResponse'

export type GenerateAdvertisementImageCandidatesResponse = {
  candidates: AdvertisementImageCandidateResponse[]
}

export const generateAdvertisementImageCandidatesResponseFromJson = (json: string): GenerateAdvertisementImageCandidatesResponse =>
  JSON.parse(json) as GenerateAdvertisementImageCandidatesResponse

export const generateAdvertisementImageCandidatesResponseToJson = (value: GenerateAdvertisementImageCandidatesResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/advertising/objects/GenerateAdvertisementImageCandidatesResponse.scala
// 本文件定义 advertising 模块的 GenerateAdvertisementImageCandidatesResponse，作为后端与前端同名的数据模型，并提供 JSON codec。
package com.typesafe.travel.advertising.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class GenerateAdvertisementImageCandidatesResponse(
    candidates: List[AdvertisementImageCandidateResponse]
)

object GenerateAdvertisementImageCandidatesResponse:
  given sourceEncoder: Encoder[GenerateAdvertisementImageCandidatesResponse] = deriveEncoder
  given sourceDecoder: Decoder[GenerateAdvertisementImageCandidatesResponse] = deriveDecoder

